### PR TITLE
Wm3 add crud projects

### PR DESCRIPTION
**Pull request by Pavel-An**
Added CRUD actions for Projects

### CODE REVIEW

File: app/controllers/projects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  def new
+    @project = current_user.projects.new
+  end
+
+  def create
+    @project = current_user.projects.new(project_params)
+
+    if @project.save
+      flash[:success] = "Project create"
```
Comment (fooooxmr) [inline app/controllers/projects_controller.rb:16]: Project created*

File: app/controllers/projects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      @project.members.create(user_id: current_user.id)
+      redirect_to @project
+    else
+      flash.now[:danger] = "Project don\'t create"
```
Comment (fooooxmr) [inline app/controllers/projects_controller.rb:20]: Project not created*

File: app/controllers/projects_controller.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def show
+    @project = Project.find(params[:id])
```
Comment (fooooxmr) [inline app/controllers/projects_controller.rb:26]: You can move `@project = Project.find(params[:id])` to the `before_action` callback. This code will be executed before the action.
```
before_action :find_project, only: [:show, :edit, :update, :destroy] #add any actions you need

...

def find_project
  @project = Project.find(params[:id])
end
```

File: app/controllers/projects_controller.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+  def edit
+    @project = Project.find(params[:id])
+  end
+
+  def update
+    @project = Project.find(params[:id])
+
+    if @project.update(project_params)
+      flash[:success] = "Project update"
```
Comment (fooooxmr) [inline app/controllers/projects_controller.rb:37]: Project updated*

**Review by fooooxmr — CHANGES_REQUESTED** [review]

File: app/controllers/projects_controller.rb (hunk 3 above)
Comment (fooooxmr) [reply app/controllers/projects_controller.rb:26]: https://rusrails.ru/action-controller-overview#kolbeki-ekshna

File: app/models/project.rb [hunk 5]
```diff
@@ -1,10 +1,9 @@
 class Project < ApplicationRecord
-  enum status: { active: 1, archived: 2 }
+  enum status: { active: 1, close: 2, paused: 3, archived: 4 }
```
[the remark this answers is not in the document]
Comment (fooooxmr) [reply app/models/project.rb:2]: Also, usually, statuses starting from 0 and not 1.

File: app/controllers/projects_controller.rb (hunk 2 above)
Comment (Pavel-An) [reply app/controllers/projects_controller.rb:20]: Done

File: app/controllers/projects_controller.rb (hunk 3 above)
Comment (Pavel-An) [reply app/controllers/projects_controller.rb:26]: Done

File: app/controllers/projects_controller.rb (hunk 4 above)
Comment (Pavel-An) [reply app/controllers/projects_controller.rb:37]: Done

File: app/controllers/projects_controller.rb [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      redirect_to @project
+    else
+      flash.now[:danger] = "Project don\'t create"
```
[the remark this answers is not in the document]
Comment (Pavel-An) [reply app/controllers/projects_controller.rb:40]: Done

File: app/models/project.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (Pavel-An) [reply app/models/project.rb:2]: Done

File: db/migrate/20241023060425_create_project_members.rb [hunk 7]
```diff
@@ -1,12 +0,0 @@
-class CreateProjectMembers < ActiveRecord::Migration[7.2]
```
[the remark this answers is not in the document]
Comment (Pavel-An) [reply db/migrate/20241023060425_create_project_members.rb:1]: Ok

**Comment by Pavel-An** [comment]
Update branch

**Review by fooooxmr — APPROVED** [review]

**Pull request merged**